common/api-review/app.api.md

```diff
@@ -119,6 +119,11 @@ export function initializeServerApp(options: FirebaseOptions | FirebaseApp, conf
 // @internal (undocumented)
 export function _isFirebaseApp(obj: FirebaseApp | FirebaseOptions): obj is FirebaseApp;
 
+// Warning: (ae-forgotten-export) The symbol "FirebaseServerAppImpl" needs to be exported by the entry point index.d.ts
```

Comment:
If you're exporting this out of `@firebase/app` you should also export `FirebaseServerAppImpl`, as its typing depends on that type, otherwise you could run into some TS errors. The other alternative is to be a little looser with the typings and not do the `obj is FirebaseServerAppImpl` and just do `boolean`.

Reply:
Yeah, I'm confused by this error. It _is_ marked as exported in `app/src/public-types.ts`. Is there somewhere else I should `export` it?

Reply:
No, to be clear, `FirebaseServerAppImpl` isn't being exported at the moment, as far as I can tell, which is what api-extractor is warning you about. `_isFirebaseServerAppImpl` is being exported. But since the typing of `_isFirebaseServerAppImpl` depends on `FirebaseServerAppImpl`, api-extractor is complaining that you have to export that too. The way to get around it is to not use the `obj is` return type and return a boolean instead.

The internal.ts file isn't "public" in that it's not meant to be used be users of the SDK, it's meant to export "internal" methods used by other packages, like auth, so I think this is fine to export this function in that file (if you check `dist/` after building you should see it show up in the "internal" bundle and not in the main bundle). The only problem is that there may possibly be some TS issue at some point if you don't resolve this warning, so I think the simplest way is to change the return type to boolean.

Reply:
This should be fixed now. I've updated the function to simple by `isFirebaseServerApp` and not `isFirebaseServerAppImpl`. `FirebasServerApp` is already a public interface so we should be good to go here.